beacon-chain/sync/receive_block.go

```diff
@@ -120,6 +124,8 @@ func (rs *RegularSync) receiveBlock(msg p2p.Message) error {
 	// and call receiveBlock recursively. The recursive function call will stop once
 	// the block we process no longer has children.
 	if child, ok := rs.hasChild(blockRoot); ok {
+		// We clear the block root from the pending processing map.
+		rs.clearPendingBlock(blockRoot)
```

Comment:
This fixes a major bug in parents processing where we would delete a pending key and then check if it existed in the map afterwards (which is a big fail)